packages/cli/src/commands/dev.ts

```diff
@@ -359,6 +362,95 @@ async function refreshEndpoint(apiClient: TriggerApi, endpointId: string, endpoi
   }
 }
 
+async function getOutdatedTriggerDevPackages(path: string) {
+  const packageJSONPath = pathModule.join(path, "package.json");
+  const packageJSON = await readJSONFile(packageJSONPath);
+
+  const packageJSONSchema = z.object({
+    dependencies: z.record(z.string()),
+  });
+
+  const triggerDevPackages = [];
+  const { dependencies } = packageJSONSchema.parse(packageJSON);
+
+  for (const packageName in dependencies) {
+    if (packageName.startsWith("@trigger.dev/")) {
+      triggerDevPackages.push(packageName);
+    }
+  }
+
+  return triggerDevPackages;
+}
+
+const packageRegistrySchema = z.object({
+  "dist-tags": z.object({
+    latest: z.string(),
+  }),
+});
+
+// @ts-ignore
+async function getLatestPackageVersion(packageName: string) {
+  const npmRegistryURL = `https://registry.npmjs.org/${packageName}`;
+
+  try {
+    const response = await fetch(npmRegistryURL);
+
+    if (response.ok) {
+      const data = await response.json();
+      const parsedData = packageRegistrySchema.safeParse(data);
+
+      if (parsedData.success) {
+        return parsedData.data["dist-tags"].latest;
+      }
+    }
+  } catch (error) {}
+
+  console.log(`Failed to fetch the lastest version of ${packageName}`);
+}
+
+async function checkForOutdatedPackages(path: string) {
+  const resolvedPath = resolvePath(path);
+  const packageJSONPath = pathModule.join(path, "package.json");
+  const packageJSON = await readJSONFile(packageJSONPath);
+
+  const outdatedPackages = [];
+  const triggerDevPackages = await getOutdatedTriggerDevPackages(resolvedPath);
+
+  for (const packageName of triggerDevPackages) {
+    const latestVersion = await getLatestPackageVersion(packageName);
+    const installedVersion =
+      packageJSON.dependencies?.[packageName] || packageJSON.devDependencies?.[packageName];
+
+    // if latestVersion isn't undefined
+    // format the versions if they include special characters like "^", so semver can parse them
+    if (latestVersion) {
+      const formtattedInstalledVersion = semver.clean(installedVersion);
+      const formattedLatestVersion = semver.clean(latestVersion);
```

Comment:
This code doesn't work with `^`.

"^2.0.8" is being turned into `null` which means that no packages are ever considered outdated.

Follow this guide on testing the CLI: https://github.com/triggerdotdev/trigger.dev/blob/main/CONTRIBUTING.md#testing-cli-changes

Test:
1. That it doesn't initially identify any outdated packages (i.e. does nothing).
2. Modify one of the versions to be older, then check if identifies it and logs it out

Reply:
@matt-aitken 

First off... if we're to modify the package version by editing the string to include `^`, with what I've observed it doesn't change the package version unless the developer does `npm install` again. 

And that means we may just be comparing the string values, and sometimes, before doing `npm install` the dev command will return that the packages (outdated and latest) are outdated as seen in this [comment](https://github.com/triggerdotdev/trigger.dev/pull/270#issuecomment-1674443817)

But, with semver, since that is what I used, all package with their key value pairs (i.e. the name and versions) are most times represented with the leading `^` and semver considers those strings as invalid package versions because I'm trying to compare them with their respective string values from the npm API response. 

Secondly, I'm sort of away from my computer for the next three weeks, I won't be able to access my PC till September 6 or thereafter. 

So, if that's okay, I'd still love to work on this issue by then, if my approach with semver doesn't quite fit the problem.

Reply:
Hey @kaf-lamed-beyt,

Thanks for this, that makes sense. Now I understand why more frameworks don't do what we're trying to do here, it's not as simple as I thought it would be…

I think a better approach is to use this package: https://www.npmjs.com/package/npm-check-updates

It has a lot of contributors and high usage and allows programmatic usage (see the bottom of the readme) with wildcards.

If run from the command line:
```
npx npm-check-updates "/trigger.dev\/.+$/"
```

It correctly outputs
![CleanShot 2023-08-18 at 11 42 33](https://github.com/triggerdotdev/trigger.dev/assets/10635986/1c72a468-8e12-468e-9d67-2e32917a98cf)

When you're back if you can make this change we'll tip you generously because we've messed you around a bit here.

Reply:
Oh yeah! I've seen `npm-check-updates` in action – frequently in some of the projects I use.

I'll be happy to implement this when I'm back. Thank you!